delta/plugins/storage/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/storage/storages/operations/s3/S3StorageLinkFile.scala

```diff
@@ -0,0 +1,55 @@
+package ch.epfl.bluebrain.nexus.delta.plugins.storage.storages.operations.s3
+
+import akka.actor.ActorSystem
+import akka.http.scaladsl.model.Uri
+import akka.stream.alpakka.s3.scaladsl.S3
+import akka.stream.alpakka.s3.{S3Attributes, S3Exception}
+import akka.stream.scaladsl.Sink
+import ch.epfl.bluebrain.nexus.delta.plugins.storage.files.model.FileAttributes.FileAttributesOrigin.Storage
+import ch.epfl.bluebrain.nexus.delta.plugins.storage.files.model.{FileAttributes, FileDescription}
+import ch.epfl.bluebrain.nexus.delta.plugins.storage.storages.model.Storage.S3Storage
+import ch.epfl.bluebrain.nexus.delta.plugins.storage.storages.operations.StorageFileRejection.FetchFileRejection
+import ch.epfl.bluebrain.nexus.delta.plugins.storage.storages.operations.{LinkFile, SaveFile, StorageFileRejection}
+import monix.bio.IO
+
+import java.net.URLDecoder
+import java.nio.charset.StandardCharsets.UTF_8
+import scala.concurrent.Future
+
+class S3StorageLinkFile(storage: S3Storage)(implicit as: ActorSystem) extends LinkFile {
+
+  import as.dispatcher
+
+  private val fileNotFoundException = new IllegalArgumentException("File not found")
+
+  override def apply(key: Uri.Path, description: FileDescription): IO[StorageFileRejection, FileAttributes] = {
+    val attributes    = S3Attributes.settings(storage.value.toAlpakkaSettings)
+    val location: Uri = s"${storage.value.address(storage.value.bucket)}/$key"
```

Comment:
why is used the string conversion here and not the actual Uri conversion?
```scala
import ch.epfl.bluebrain.nexus.delta.rdf.syntax.sdk.syntax._

val location: Uri = storage.value.address(storage.value.bucket) / key
```

Reply:
I forgot about it